### PR TITLE
Valencia nonatty

### PR DESCRIPTION
**Pull request by oxtoacart**
This PR removes references from flashlight to natty, waddell and all the P2P stuff.  The projects are still kept in flashlight-build for eventual re-integration, but the produced binary doesn't contain them.

### CODE REVIEW

File: src/github.com/getlantern/flashlight/nattest/nattest.go [hunk 1]
```diff
@@ -1,148 +0,0 @@
-// package nattest provides the capability to test a nat-traversed UDP
```
Comment (atavism) [inline src/github.com/getlantern/flashlight/nattest/nattest.go:1]: I think testpackages.txt needs updated now that this has been removed?

**Comment by atavism** [comment]
I'm seeing 

```
Stderr: # github.com/getlantern/systray
../systray/systray.m:75:22: warning: sending 'TrayMenu *' to parameter of incompatible type 'id<NSFileManagerDelegate>'
/System/Library/Frameworks/Foundation.framework/Headers/NSFileManager.h:109:47: note: passing argument to parameter 'delegate' here
# github.com/getlantern/flashlight
src/github.com/getlantern/flashlight/flashlight.go:45: main redeclared in this block
    previous declaration at src/github.com/getlantern/flashlight/flashlight.go:14
src/github.com/getlantern/flashlight/flashlight.go:46: undefined: systray
src/github.com/getlantern/flashlight/flashlight.go:177: undefined: systray
src/github.com/getlantern/flashlight/flashlight.go:178: undefined: systray
src/github.com/getlantern/flashlight/flashlight.go:179: undefined: systray
```

when I try to compile on this branch. Are some changes missing? Same on the valencia branch too actually.

**Comment by fffw** [comment]
@atavism your code may not be up to date. `systray/systray.m` doesn't exist under https://github.com/getlantern/flashlight-build/tree/valencia-nonatty/src/github.com/getlantern/systray or https://github.com/getlantern/flashlight-build/tree/valencia/src/github.com/getlantern/systray

**Comment by atavism** [comment]
@fffw Thanks. Looks like I had an old version of systray installed.

**Comment by oxtoacart** [comment]
@atavism What's your go path?  When you compile flashlight under flashlight-build, it should be using the systray project inside of flashlight-build, not from somewhere else.  Are you running `source ./setenv.bash` inside flashlight-build?

**Comment by atavism** [comment]
@oxtoacart I actually set my GOPATH to the flashlight-build directory and another directory to avoid running that script every time. It works fine normally; I just didn't realize there was an old version of systray installed in the other directory. It's resolved now.

**Comment by atavism** [comment]
This LGTM besides removing the `github.com/getlantern/flashlight/nattest` line from testpackages.txt

**Comment by oxtoacart** [comment]
@atavism Does flashlight-build come first in your GOPATH?  I ask because one of the purposes of flashlight-build is having repeatable builds, but if one uses a GOPATH that gives precedence to resources outside of flashlight-build, the builds won't be repeatable.  We caught it this time because the build actually broke, but what I'm concerned about is when the build doesn't break but we're using old/buggy code for some libraries.

**Comment by oxtoacart** [comment]
> This LGTM besides removing the github.com/getlantern/flashlight/nattest line from testpackages.txt

Thanks!  Fixed.

**Pull request merged**